### PR TITLE
Assure only one initialization; Use executor for notification thread

### PR DESCRIPTION
**Pull request by tepi**

### CODE REVIEW

File: appsec-kit-v7/src/main/java/com/vaadin/appsec/v7/service/NotificationInitializer.java [hunk 1]
```diff
@@ -56,32 +63,37 @@ static void serviceInit(VaadinService vaadinService) {
...
+                        ScheduledFuture<?> thisCheck = scheduledChecks
+                                .get(sessionId);
+                        if (thisCheck != null) {
+                            thisCheck.cancel(true);
+                            scheduledChecks.remove(sessionId);
+                        }
+                    }
+                }, NOTIFICATION_CHECK_INTERVAL, NOTIFICATION_CHECK_INTERVAL,
+                        TimeUnit.MILLISECONDS);
+        scheduledChecks.put(session.getSession().getId(),
```
Comment (tamasmak) [inline appsec-kit-v7/src/main/java/com/vaadin/appsec/v7/service/NotificationInitializer.java:95]: Could be the `sessionId` used here also?

Comment (tepi) [reply appsec-kit-v7/src/main/java/com/vaadin/appsec/v7/service/NotificationInitializer.java]: Yes, thanks!

File: appsec-kit-v7/src/main/java/com/vaadin/appsec/v7/service/NotificationInitializer.java [hunk 2]
```diff
@@ -56,32 +63,36 @@ static void serviceInit(VaadinService vaadinService) {
...
-                // NOP
-            }
-        });
+    private static void startNotificationTask(final VaadinSession session) {
+        final String sessionId = session.getSession().getId();
+        final ScheduledFuture<?> scheduledNotificationCheck = AppSecService
+                .getInstance().getConfiguration().getTaskExecutor()
+                .scheduleAtFixedRate(() -> {
+                    LOGGER.info(
+                            "NotificationInitListener notification thread initialized.");
```
Comment (tamasmak) [inline appsec-kit-v7/src/main/java/com/vaadin/appsec/v7/service/NotificationInitializer.java:78]: Should we log this message each time the task runs?

Comment (tepi) [reply appsec-kit-v7/src/main/java/com/vaadin/appsec/v7/service/NotificationInitializer.java:78]: Definitely not. Good catch!

Comment (tepi) [reply appsec-kit-v7/src/main/java/com/vaadin/appsec/v7/service/NotificationInitializer.java]: Fixed

**Review by tamasmak — APPROVED** [review]
Approving this PR with a comment that there will be an another PR where the notification task will be removed and the `AppSecScanEvent` will be used to notify the UIs if new vulnerabilities are found.

**Pull request merged**